packages/@aws-cdk/aws-lambda/lib/event-source-mapping.ts

```diff
@@ -332,6 +342,27 @@ export class EventSourceMapping extends cdk.Resource implements IEventSourceMapp
       selfManagedEventSource = { endpoints: { kafkaBootstrapServers: props.kafkaBootstrapServers } };
     }
 
+    let amazonManagedKafkaEventSourceConfig : CfnEventSourceMapping['amazonManagedKafkaEventSourceConfig'];
+    let selfManagedKafkaEventSourceConfig: CfnEventSourceMapping['selfManagedKafkaEventSourceConfig'];
```

Comment:
Let's figure out which one of these it should be mapped to and just set it there instead of providing both props.

Reply:
I split it into two if statements to have less branching. Additionally i moved the validation to private function to have a cleaner view on the logic.

Reply:
I like this much better. Thanks!